sdk/appcenter/src/main/java/com/microsoft/appcenter/persistence/SessionStorage.java

```diff
@@ -0,0 +1,207 @@
+package com.microsoft.appcenter.persistence;
+
+import android.support.annotation.VisibleForTesting;
+import android.support.annotation.WorkerThread;
+
+import com.microsoft.appcenter.utils.AppCenterLog;
+import com.microsoft.appcenter.utils.storage.StorageHelper;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.UUID;
+
+import static com.microsoft.appcenter.AppCenter.LOG_TAG;
+
+/**
+ * Persistent session history.
+ */
+public class SessionStorage {
```

Comment:
This is a storage but does more than just maintaining storage. This is why I named as `SessionContext`. I just wonder if you have strong opinion on the name `SessionStorage`, otherwise we can use same name as Apple SDK.

Reply:
I can rename for platform consistency.

Reply:
Done